hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/WritableECContainerProvider.java

```diff
@@ -159,6 +160,16 @@ public ContainerInfo getContainer(final long size,
     }
   }
 
+  private int getMinimumPipelines(ECReplicationConfig repConfig) {
```

Comment:
At this code path with be hit for every single block request, I wonder if we should perform this calculation every time? It has to iterate over all nodes in the cluster checking their state and then summing up their disk count. Its probably not needed to recalculate this every single time - we could either calculate it 1 in X times, or have some sort of time based recalculation. Its also called under a synchronized block, so it could become a point of contention in the block allocation flow.